src/processFrames.js

```diff
@@ -185,6 +250,48 @@ function calculateFps(gops) {
     };
 }
 
+function calculateGopDuration(gops) {
+    const gopsDurations = [];
+
+    gops.forEach(gop => {
+        const gopDurationInSec = processFrames.gopDurationInSec(gop);
+
+        gopsDurations.push(gopDurationInSec);
+    });
+
+    return {
+        mean: _.mean(gopsDurations),
+        min: Math.min(...gopsDurations),
+        max: Math.max(...gopsDurations)
+    };
+}
+
+function calculateAspectRatio(width, height) {
+    const arCoefficient = width / height;
+
+    if (Math.abs(arCoefficient - SQUARE_AR_COEFFICIENT) <= AR_CALCULATION_PRECISION) {
+        return SQUARE_AR;
+    }
+
+    if (Math.abs(arCoefficient - TRADITIONAL_TV_AR_COEFFICIENT) <= AR_CALCULATION_PRECISION) {
+        return TRADITIONAL_TV_AR;
+    }
+
+    if (Math.abs(arCoefficient - HD_VIDEO_AR_COEFFICIENT) <= AR_CALCULATION_PRECISION) {
+        return HD_VIDEO_AR;
+    }
+
+    if (Math.abs(arCoefficient - UNIVISIUM_AR_COEFFICIENT) <= AR_CALCULATION_PRECISION) {
+        return UNIVISIUM_AR;
+    }
+
+    if (Math.abs(arCoefficient - WIDESCREEN_AR_COEFFICIENT) <= AR_CALCULATION_PRECISION) {
+        return WIDESCREEN_AR;
+    }
+
+    return `${width}:${height}`;
```

Comment:
I think that we need to calculate [GCD](https://en.wikipedia.org/wiki/Greatest_common_divisor). It's common technique in ffmpeg.

Reply:
+